module1/homework/lengthSort/sort.cpp

```diff
@@ -0,0 +1,23 @@
+#include "sort.hpp"
+
+#include <algorithm>
+
+std::deque<std::string> lengthSort(std::forward_list<std::string> list) {
+    std::deque<std::string> deque;
+    std::copy(list.begin(), list.end(), std::back_inserter(deque));
+
+    std::sort(deque.begin(), deque.end(), [](const auto& lhs, const auto& rhs) {
+        if (lhs.size() < rhs.size()) {
+            return true;
+        } else if (lhs.size() == rhs.size()) {
+            if (lhs < rhs) {
+                return true;
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
```

Comment:
You complicated here. Just check if 2 values have the same size and compare them, and if not return (true or false) if first is bigger than second.